src/extlibs/hrvo/simulator.cpp

```diff
@@ -39,112 +39,179 @@
+                }
+            }
         }
 
-        if (intersection_point_set.empty())
+        for (const Robot &enemy_robot : enemy_team)
         {
-            // If there is no intersection point (robot is outside the field), continue
-            // moving in the current direction
-            intersection_point_set.insert(enemy_robot.position() +
-                                          enemy_robot.velocity() * 5);
-        }
+            auto agent_index_iter = enemy_robot_id_map.find(enemy_robot.id());
+            if (agent_index_iter != enemy_robot_id_map.end())
+            {
+                unsigned int agent_index = agent_index_iter->second;
+                Point position           = enemy_robot.position();
+                agents[agent_index]->setPosition(Vector2(position.x(), position.y()));
 
-        Vector2 goal_position(static_cast<float>(intersection_point_set.begin()->x()),
-                              static_cast<float>(intersection_point_set.begin()->y()));
-        addLinearVelocityRobotAgent(enemy_robot, goal_position);
+                Vector velocity = enemy_robot.velocity();
+                agents[agent_index]->setVelocity(Vector2(velocity.x(), velocity.y()));
+            }
+        }
     }
 
+    // TODO (#2498): Dynamically add and remove the ball as an Agent, and if needed
+    //               update its radius based on the PrimitiveSet
     if (add_ball_agent)
     {
-        // Ball should be treated as an agent (obstacle)
-        const Ball &ball = world.ball();
-        Vector2 position(ball.position().x(), ball.position().y());
-        Vector2 velocity(ball.velocity().x(), ball.velocity().y());
-        Vector2 goal_pos   = position + 100 * velocity;
-        float acceleration = ball.acceleration().length();
-        // Minimum of 0.5-meter distance away from the ball, if the ball is an obstacle
-        float ball_radius = 0.5f + BALL_AGENT_RADIUS_OFFSET;
-
-        addLinearVelocityAgent(position, ball_radius, velocity, abs(velocity),
-                               acceleration, addGoal(goal_pos), 0.1f);
+        if (ball_agent_id == -1)
+        {
+            // Ball should be treated as an agent (obstacle)
+            const Ball &ball = world.ball();
+            Vector2 position(ball.position().x(), ball.position().y());
+            Vector2 velocity(ball.velocity().x(), ball.velocity().y());
+            Vector2 goal_pos   = position + 100 * velocity;
+            float acceleration = ball.acceleration().length();
+            // Minimum of 0.5-meter distance away from the ball, if the ball is an
+            // obstacle
+            float ball_radius = 0.5f + BALL_AGENT_RADIUS_OFFSET;
+
+            std::size_t agent_index =
+                addLinearVelocityAgent(position, ball_radius, velocity, abs(velocity),
+                                       acceleration, addGoal(goal_pos), 0.1f);
+            ball_agent_id = agent_index;
```

Comment:
I see that if we receive a `stay_away_from_ball` = True, we create this agent, but it never goes away? I could just be missing the code that handles the remove part